install_config/registry/deploy_registry_existing_clusters.adoc

```diff
@@ -91,7 +94,204 @@ Daemonsets ensure that when nodes are created, they contain copies of a
++
+====
+----
+$ oc get -n default -o yaml dc/docker-registry \ <1>
+  >dc-docker-registry.yaml
+----
+<1> Replace `dc` with `ds` if your *deployment* is a DaemonSet. The same applies to the next
+command below.
+====
++
+1. Delete the existing deployment:
++
+----
+$ oc delete -n default dc/docker-registry
+----
++
+2. Create a new deployment with `oc adm registry ...`:
++
+The command will fail because some objects already exist as the following output illustrates. That
+can be ignored because the desired DeploymentConfig has been successfully created.
++
+====
+----
+$ oc adm registry -o json \ <1>
+  | oc create -f -
+deploymentconfig "docker-registry" created
+Error from server (AlreadyExists): serviceaccounts "registry" already exists
+Error from server (AlreadyExists): clusterrolebindings "registry-registry-role" already exists
+----
+<1> Add additional arbitrary arguments as you see fit. Add `*--daemonset*` if you want to deploy
+a DaemonSet.
+====
+
+=== Changing the service IP
+
+xref:maintaining-the-registry-ip-address[Should be avoided] if possible. However, it can be useful
+when the service has been already unintentionally reinstantiated with a different IP while there
+are existing objects in the registry. The process is different for the DeploymentConfig and
+DaemonSet.
+
+DeploymentConfig::
+
+====
+1. Replace the existing service object with new one with the desired cluster IP:
++
+----
+$ oc expose -n default dc/docker-registry \ <1>
+  --session-affinity=ClientIP --cluster-ip=172.30.1.1 <2> \
+  -o json --dry-run | \
+  oc replace -n default -f - --force <3>
+----
++
+2. xref:#restart-masters[Restart the masters].
++
+3. Restart the registry pods:
++
+This is needed unless `*REGISTRY_OPENSHIFT_SERVER_ADDR*` is set for the registry services to
+reload their internal registry address.
++
+----
+$ oc rollout latest -n default dc/docker-registry
+----
+
+<1> The command produces a specification of the new service named `docker-registry` in the default
+namespace for the given DeploymentConfig.
+<2> Use the desired IP address instead.
+<3> If the service does not exist anymore, change the last line to `oc create -f -`.
+====
+
+DaemonSet::
+
+In case of a DaemonSet, its update strategy needs to be first changed to `*RollingUpdate*` so that
+pods will be cleanly restarted in a controlled fashion. 
+
+====
+1. Set the update strategy to `*RollingUpdate*`:
++
+----
+$ oc patch -n default ds/docker-registry \
+  -p '{"spec":{"updateStrategy":{"type":"RollingUpdate"}}}'
+----
++
+2. Replace the registry's service:
++
+----
+$ oc create service clusterip -n default docker-registry --dry-run --tcp=5000:5000 \
+  -o json --clusterip=172.30.1.1 | \ <1>
+  oc patch -f - --type=json -p '[
+      {"op":"add","path":"/spec/sessionAffinity","value":"ClientIP"},
+      {"op":"add","path":"/spec/selector","value":{"docker-registry":"default"}}
+    ]' --local=true -o json | \
+    oc replace -n default -f - --force <2>
```

Comment:
Couldn't find out anything simpler. Originally I wanted to use `oc adm registry` and filter out everything but the `service` template. But I couldn't pull it off without using external tools like `jq`.